src/components/interaction-designer/toolbar/ErrorNotifications.vue

```diff
@@ -123,9 +123,17 @@ export class ErrorNotifications extends mixins(Routes, Lang) {
 
   get invalidBlocksInActiveFlow() {
     return filter(this.activeFlow?.blocks, (block) => {
-      // The block is invalid on IBlock OR it's invalid on corresponding IResource
-      return get(this.blockValidationStatuses, `block/${block.uuid}`) || get(this.resourceValidationStatusesForActiveFlow, `resource/${block.config.prompt}`)
-    });
+      const hasBlockValidationError = has(this.blockValidationStatuses, `block/${block.uuid}`)
+      const hasResourceValidationError = has(this.resourceValidationStatusesForActiveFlow, `resource/${block.config.prompt}`)
+      let hasChoiceValidationError = false
+
+      if (block.config?.choices !== undefined) {
+        hasChoiceValidationError = Object.values(block.config.choices)
+          .reduce((result, uuid) => (result === true || has(this.resourceValidationStatusesForActiveFlow, `resource/${uuid}`)), false)
+      }
+
+      return hasBlockValidationError || hasResourceValidationError || hasChoiceValidationError
```

Comment:
Let's make an `early return` pattern for this. Pseudo code:
```
if(hasBlockValidationError) {
   return true
} else if (hasResourceValidationError) {
   return true
}

// continue other instructions 
return hasChoiceValidationError
```

This will avoid us to always run the `hasChoiceValidationError = Object.values(block.config.choices)
          .reduce((result, uuid) => (result === true || has(this.resourceValidationStatusesForActiveFlow, `resource/${uuid}`)), false)`

Reply:
I tried using short-circuit to optimize execution